anospp_analysis/nn.py

```diff
@@ -0,0 +1,501 @@
+                                                colors_int, colors_fine]):
+            fig, _ = plot_assignment_proportions(result, level, colors, args.nn_assignment_threshold)
+            fig.savefig(f'{args.outdir}/{level}_assignment.png')
+
+    logging.info('All done!')
+
+    
+def main():
+    
+    parser = argparse.ArgumentParser("NN assignment for ANOSPP sequencing data")
+    parser.add_argument('-a', '--haplotypes', help='Haplotypes tsv file', required=True)
+    parser.add_argument('-m', '--manifest', help='Samples manifest tsv file', required=True)
+    parser.add_argument('-s', '--stats', help='DADA2 stats tsv file', required=True)
+    parser.add_argument('-r', '--reference', help='Reference index version - currently a directory name.\
+         Default: nnv1', default='nnv1')
+    parser.add_argument('-p', '--path_to_refversion', help='path to reference index version.\
+         Default: test_data', default='test_data')
+    parser.add_argument('--no_plotting', help='Do not generate plots. Default: False', \
+                        default=False)
+    parser.add_argument('--allelism_normalisation', help='Normalisation method over multiple alleles. Options: \
+                         [n_alleles,reads_fraction]. Default: n_alleles', default='n_alleles')
+    parser.add_argument('--hap_read_count_threshold', help='minimum number of reads for supported haplotypes. \
+         Default: 10', default=10)
+    parser.add_argument('--hap_reads_fraction_threshold', help='minimum fraction of reads for supported haplotypes. \
+         Default: 0.1', default=0.1)
+    parser.add_argument('--medium_contamination_read_count_threshold', help='samples with fewer than this number \
+                        of reads get medium contamination risk. Default: 1000', default=1000)
+    parser.add_argument('--medium_contamination_multi_allelic_threshold', help='samples with more than this number \
+                        of multiallelic targets get medium contamination risk. Default: 0', default=0)
+    parser.add_argument('--high_contamination_multi_allelic_threshold', help='samples with more than this number \
+                        of multiallelic targets get high contamination risk. Default: 2', default=2)
+    parser.add_argument('--nn_assignment_threshold', help='required fraction for calling assignment. \
+                        Default: 0.8', default=0.8)
+    parser.add_argument('--n_error_snps', help='Maximum number of snps for a multi-allelic sequence to be \
+                        considered a sequencing or PCR error. Default: 2', default=2)
+    parser.add_argument('-k', '--kmer_length', help='Length of k-mers to use. Note that NNoVAE has been developed \
+                        and tested for k=8, so accuracy of results cannot be guaranteed with other values of k. \
+                        Default: k=8', default=8)
+    parser.add_argument('-o', '--outdir', help='Output directory. Default: nn', default='nn')
```

Comment:
just for clarity, can we move outdir arg just above no_plotting?

Reply:
yes.